core/src/main/java/org/jahia/modules/modulemanager/flow/ModuleManagementFlowHandler.java

```diff
@@ -197,8 +198,16 @@ private boolean handleModule(File file, MessageContext context, String originalF
 
             //If it is not a jar and none impl can handle this extension
             if (!canHandleExtension) {
-                context.addMessage(new MessageBuilder().error().source("moduleFile")
-                        .code("serverSettings.manageModules.install.wrongFormat").build());
+                MessageBuilder moduleFileMsgBuilder = new MessageBuilder().error().source("moduleFile");
+                // Use a custom error message for tgz files when no Javascript Modules Engine is installed
+                if (FilenameUtils.isExtension(fileNameLoweredCase, "tgz")) {
+                    context.addMessage(moduleFileMsgBuilder
```

Comment:
A bit sad to have a 'if:tgz' in this code that was for now generic enough to not have specific cases (thx to the AritfactUrlTransformer and OSGI implems).
But I don't see any other alternative solution, since the engine module is not present when such case is happening.
So, let's go for it.

Reply:
I agree, this solution is not ideal but I couldn't find a cleaner fix :/